nexus/lib/nexus/peer.ex

```diff
@@ -84,21 +120,20 @@ defmodule Nexus.Peer do
 
     outbound_tracks = setup_transceivers(pc, peer_ids)
 
-    state = State.send_offer(state)
+    state = send_offer(state)
 
     {:noreply, %{state | outbound_tracks: outbound_tracks}}
   end
 
   @impl true
-  def handle_call({:apply_sdp_answer, answer_sdp}, _from, state) do
+  def handle_call({:apply_sdp_answer, answer_sdp}, _from, %{pc: pc} = state) do
     state =
-      if State.offer_pending?(state) do
+      if PeerConnection.get_signaling_state(pc) == :have_local_offer do
         state
-        |> State.apply_answer(answer_sdp)
-        |> State.flush_candidates()
-        |> State.send_notifications()
+        |> apply_answer(answer_sdp)
+        |> send_notifications()
       else
-        Logger.debug("Ignoring SDP answer for #{state.id}: no pending offer")
+        Logger.warning("Ignoring SDP answer for #{state.id}: no pending offer")
```

Comment:
I'd probably just apply the answer and log the warning in case it returned an error, but that's a nitpick

Reply:
I thought applying an answer in signaling state `:stable` led to a crash?

Reply:
It should return `{:error, :invalid_state}`, it it crashes, that's a bug.

Reply:
Confirming it behaves as you described, I must've remembered incorrectly